gammapy/irf/background.py

```diff
@@ -72,6 +72,18 @@ def __str__(self):
         ss += '\n{}'.format(self.data)
         return ss
 
+    @property
+    def energy(self):
```

Comment:
@thomasarmstrong - How do you feel about re-exposing the axes as properties on the IRF class?

My suggestion would be to remove them, and instead to show in the class-level docstring an example how to obtain the axes via the `data` attribute (the `NDDataArray`). My reasoning is that it's better to start minimal, because it's better to have little code and adding is always possible in the future, but removing causes anger after people (or e.g. code in ctapipe) starts using it. I know at the moment some of our IRF classes have them re-exposed and some don't, so if you agree with this suggestion, please remove them from all IRF classes here.

If you think it's useful to re-expose them: please add a one-line docstring to each property, so that it shows up in a good way in the API docs. I.e. something like this:
```
Energy axis (`~gammapy.utils.nddate.BinnedDataAxis`).
```

Maybe @kosack also has an opinion here?

Reply:
@cdeil - so the idea would be to remove all the @propery definitions for energy, offset etc for all the IRF classes, and replace things like
```
energy = self.energy.bins
```
with
```
energy = self.axis('energy').bins
```
For example in EffectiveAreaTable2D.to_effective_area_table?

Reply:
Yes. But I don't have a strong preference here, so you can also go with what you prefer here.

Reply:
that would be beneficial later if we have add more axes than just energy, offset, etc.